nautobot/docs/models/extras/jobbutton.md

```diff
@@ -0,0 +1,166 @@
+        # Add job logic here
+```
+
+### Job Buttons for Multiple Types
+
+Since Job Buttons can be associated to multiple object types, it would be trivial to create a Job that can change what it runs based on the object type.
+
+```py
+from nautobot.dcim.models import Device, Site
+from nautobot.extras.jobs import JobButtonReceiver
+
+
+class ExampleComplexJobButtonReceiver(JobButtonReceiver):
+    class Meta:
+        name = "Example Complex Job Button Receiver"
+
+    def _run_site_job(self, obj):
+        self.log_info(obj=obj, message="Running Site Job Button Receiver.")
+        # Run Site Job function
+
+    def _run_device_job(self, obj):
+        self.log_info(obj=obj, message="Running Device Job Button Receiver.")
+        # Run Device Job function
+
+    def receive_job_button(self, obj):
+        if isinstance(obj, Site):
+            self._run_site_job(obj)
+        elif isinstance(obj, Device):
+            self._run_device_job(obj)
+        else:
+            self.log_failure(obj=obj, message=f"Unable to run Job Button for type {type(obj).__name__}.")
+```
+
+### Add a Job Button to Existing Jobs
+
+As was explained above, Job Buttons may only be associated to JobButtonReceiver subclasses. If you have an existing Job that would work as a Job Button, you can either rewrite it to be a JobButtonReceiver (as above) or you can move the existing logic to a callable that each Job subclass would call. This would allow the existing Job to remain while also adding the Job Button functionality.
+
+```py
+from nautobot.dcim.models import Device, Site
+from nautobot.extras.jobs import Job, ObjectVar, JobButtonReceiver
+
+
+class ExampleSiteObjectJob(Job):
+    site = ObjectVar(model=Site)
+
+    class Meta:
+        name = "Example Site Object Job"
+
+    def run(self, data, commit):
+        site = data["site"]
+        if commit:
+            _run_site_job(self, site)
+
+
+class ExampleDeviceObjectJob(Job):
+    device = ObjectVar(model=Device)
+
+    class Meta:
+        name = "Example Device Object Job"
+
+    def run(self, data, commit):
+        device = data["device"]
+        if commit:
+            _run_device_job(self, device)
+
+
+class ExampleJobButtonReceiverOtherJobs(JobButtonReceiver):
+    class Meta:
+        name = "Example Job Button Receiver for Other Jobs"
+
+    def receive_job_button(self, obj):
+        if isinstance(obj, Site):
+            _run_site_job(self, obj)
+        elif isinstance(obj, Device):
+            _run_device_job(self, obj)
+        else:
+            self.log_failure(obj=obj, message=f"Unable to run Job Button for type {type(obj).__name__}.")
+
+
+def _run_site_job(job_class, obj):
+    job_class.log_info(obj=obj, message="Running Site Job.")
+    # Add Site job logic here
+
+
+def _run_device_job(job_class, obj):
+    job_class.log_info(obj=obj, message="Running Device Job.")
+    # Add Device job logic here
+```
+
+## Context Data
```

Comment:
I am not seeing the context data in any of the examples above? To me that would have broken the signature, so I am not sure I am fully tracking.

Reply:
I _think_ @joewesch this is so you could do something like `JobButton(text="Check {obj.name}")`, right?

Reply:
Spoke with @joewesch, I had it backwards, the context of user and similar is already in `self`, Joe is kind enough to add an example of that, but no coding change requested once I understood.

Reply:
Added in latest commits.